torch/csrc/distributed/c10d/NanCheck.cu

```diff
@@ -73,6 +78,40 @@ struct CheckBytePack<T, /*EltPerPack*/8> {
   }
 };
 
+// (v) Template specialization for Float8_e4m3fn.
+// EltPerPack = 16 / 1 = 16
+
+// isnan condition for Float8_e4m3fn:
+// (x & 0b01111111) == 0b01111111
+// i.e.
+// (x & 0x7f) == 0x7f
+
+// We want to check 8 x FP8 simultaneously, i.e. at size of 8 bytes.
+// The algorithm is as follows:
+// (1) Mask out the most significant bit with mask 0x7f.
+// (2) If the result is 0x7f (is nan), then the following would be a negative number:
+//     x[i] = 0x7f - (x[i] + 0x01)
+// (3) Only leave the sign bit by masking with 0x80.
+// (4) If any x[i] is negative, then the whole x != 0.
+#define CHECK_8_FLOAT8_E4M3FN(x) \
```

Comment:
I suggest making this a function/template specialization (e.g. has_nan_vectorized) for type safety. Also it would help organizing similar optimizations for other scalar types.

Reply:
Yep, changed it to inline function now.